kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -642,10 +666,15 @@ KeysVector DBAdapter::genesisBlockKeyDeletes(BlockId blockId) const {
     keysToDelete.push_back(DBKeyManipulator::extractKeyFromStaleKey(staleKey));
   }
 
-  // TODO(DD): Iterate over the NonProvableStale keys and add them to keysToDelete.
-
   // Clear the stale index for the last version.
   add(keysToDelete, staleKeys);
+  const auto &nonProvableKeys = staleIndexKeysForBlock(blockId);
+
+  // Clear the non-provable stale index for the last version.
```

Comment:
I think that comment only relates to the `add()` call. Maybe we can be more explicit and separate the lines in a way it is clearer.